packages/ember-application/lib/system/application.js

```diff
@@ -256,28 +256,27 @@ var Application = Namespace.extend(DeferredMixin, {
   init: function() {
     this._super();
 
-    // Start off the number of deferrals at 1. This will be
-    // decremented by the Application's own `initialize` method.
-    this._readinessDeferrals = 1;
-
     if (!this.$) {
       this.$ = jQuery;
     }
 
-    // Create subclass of Ember.Router for this Application instance.
-    // This is to ensure that someone reopening `App.Router` does not
-    // tamper with the default `Ember.Router`.
-    // 2.0TODO: Can we move this into a globals-mode-only library?
-    this.Router = Router.extend();
     this.buildRegistry();
 
-    // TODO:(tomdale+wycats) Move to session creation phase
-    this.buildInstance();
-
     registerLibraries();
     logLibraryVersions();
 
-    this.scheduleInitialize();
+    // Start off the number of deferrals at 1. This will be
+    // decremented by the Application's own `initialize` method.
+    this._readinessDeferrals = 1;
+
+    if (true) {
+      // Create subclass of Ember.Router for this Application instance.
+      // This is to ensure that someone reopening `App.Router` does not
+      // tamper with the default `Ember.Router`.
+      // 2.0TODO: Can we move this into a globals-mode-only library?
+      this.Router = Router.extend();
+      this.waitForDOMReady(this.buildDefaultInstance());
```

Comment:
Would be great to not wait for DOM ready, especially when you [stream a chunked encoded](http://zoompf.com/blog/2012/05/too-chunky) index.html.  We optimize by streaming down JSON snippets to the index.html. Would be great to hide Ember boot underneath the initial connection.